### ISSUE TITLE
Game crashes after entering world 

### ISSUE THREAD

**Issue by BugmanBugman** [MarkusBordihn/adaptive_performance_tweaks#24]
**Describe the bug**
Upon generating a world, and exploring it for a bit,  the game crashes

**To Reproduce**
1. create world
2. explore a bit
3. expect crash

**Expected behavior**
To nto crash

**Debugging**
[crash-2021-08-21_12.29.21-server.txt](https://github.com/MarkusBordihn/adaptive_performance_tweaks/files/7025928/crash-2021-08-21_12.29.21-server.txt)

**Comment by MarkusBordihn** [comment]
Thanks for the report. Unfortunately I'm not able to reproduce the issue even after running and test instance for 2 hours and automatic moving around.

Furthermore the **setViewDiestnace** feature is using the default forge function, so it's not using chunkholder or and custom modification at all.
According the log I'm assume that some of the used core-mods / mixin causing this issue like:

- lithium.mixins.json:alloc.chunk_ticking.ServerChunkManagerMixin
- lithium.mixins.json:world.chunk_access.ServerChunkManagerMixin
- lithium.mixins.json:world.chunk_ticking.ThreadedAnvilChunkStorageMixin
- lithium.mixins.json:world.player_chunk_tick.ThreadedAnvilChunkStorageMixin

Could you please share a link to the lithium version you are using so that I could test this more in detail.

Normally you should get a warning in the log if you are using core-mods / mixin which could cause issues.

**Comment by BugmanBugman** [comment]
https://github.com/AeiouEnigma/lithium-forge/releases

**Comment by MarkusBordihn** [comment]
Thanks for the link. Unfortunately I was not really able to reproduce the issue with different kind of settings with only these two mods.

If you are able to reproduce the issue regular, could you please test the following:

- Does is work with the latest APT version (1.1.3) and Lithium (Forge) ?
- Does ist work with the latest APT version (1.1.3) without Lithium (Forge) ?
- Does Single Player / Multi Player makes any different ?
- What is visible in the normal log before the crash ?

Thanks a lot.

**Comment by Cynix32** [comment]
Similar issue, but instead i get a ticking memory exception error. I thought this was caused by sodium reforged since it also incorporates lithium, so i removed it. Problem still persists. Removing APTweaks solved the problem. I do not have the time to thoroughly test it since univ has already started.
https://pastebin.com/Sd2bjK14

**Comment by Cynix32** [comment]
Additional info
MC 1.16.5
Forge 36.2.4
APT 1.16.5-1.1.3

**Comment by MarkusBordihn** [comment]
@Cynix32 Thanks for the additional report. It seems somehow related, but by back tracking the mapping they all point to an error in your chunk data of the world. I assume that increasing / deceasing the view distance over the client limit cause somehow a edge case.

I will continue to try to reproduce the issue on my side. But at the moment I could only recommend to turn of the `optimizePlayerLogin` in the config file to see if this solves the issue.

**Comment by Cynix32** [comment]
Idk why, but 
1. During that report I made, the world loaded without APTweaks, so I played a little bit.
2. I exited minecraft, then re-activated APTweaks
3. I expected it to crash again since I re-activated APTweaks
4. Somehow minecraft did not crash again on world load. note that I did not change anything in the config yet

I have zero clue as to why did this happen. I assume it has something to do with chunk data somehow getting "fixed" when I managed to load the world successfully without APTweaks, then saved the new chunk data. I think that there was an edge case wherein my chunk data got corrupted, and that corruption caused a weird interaction with APTweaks, causing it to crash on world load.

Thought I should report this since its kinda absurd that the fix is simply to temporarily deactivate APTweaks, load world, then restart minecraft with APTweaks re-activated.
